pkgs/applications/misc/deadd-notification-center/default.nix

```diff
@@ -34,13 +34,16 @@ stdenv.mkDerivation rec {
   ];
 
   installPhase = ''
-    mkdir -p $out/bin $out/share/dbus-1/services
+    mkdir -p $out/bin $out/lib/systemd $out/share/dbus-1/services
 
     cp $src/.out/${pname} $out/bin/
     chmod +x $out/bin/${pname}
 
-    sed "s|##PREFIX##|$out|g" $src/${pname}.service.in > \
+    sed "s|##PREFIX##|$out|g" $src/com.ph-uhl.deadd.notification.service.in > \
       $out/share/dbus-1/services/com.ph-uhl.deadd.notification.service
+
+    sed "s|##PREFIX##|$out|g" $src/${pname}.service.in > \
+      $out/lib/systemd/${pname}.service
   '';
```

Comment:
Why not use the `make install` in the package, using something like:

```patch
diff --git a/pkgs/applications/misc/deadd-notification-center/default.nix b/pkgs/applications/misc/deadd-notification-center/default.nix
index 35627b28d15..839a3228a0b 100644
--- a/pkgs/applications/misc/deadd-notification-center/default.nix
+++ b/pkgs/applications/misc/deadd-notification-center/default.nix
@@ -19,8 +19,6 @@ stdenv.mkDerivation rec {
     sha256 = "QaOLrtlhQyhMOirk6JO1yMGRrgycHmF9FAdKNbN2TRk=";
   };
 
-  dontUnpack = true;
-
   nativeBuildInputs = [
     autoPatchelfHook
     wrapGAppsHook
@@ -33,15 +31,18 @@ stdenv.mkDerivation rec {
     hicolor-icon-theme
   ];
 
-  installPhase = ''
-    mkdir -p $out/bin $out/share/dbus-1/services
-
-    cp $src/.out/${pname} $out/bin/
-    chmod +x $out/bin/${pname}
+  buildFlags = [
+    # Exclude stack from `make all` to use the prebuilt binary from `.out/`.
+    "service"
+  ];
 
-    sed "s|##PREFIX##|$out|g" $src/${pname}.service.in > \
-      $out/share/dbus-1/services/com.ph-uhl.deadd.notification.service
-  '';
+  makeFlags = [
+    "PREFIX=${placeholder "out"}"
+    "SERVICEDIR_SYSTEMD=${placeholder "out"}/etc/systemd/user"
+    "SERVICEDIR_DBUS=${placeholder "out"}/share/dbus-1/services"
+    # Override systemd auto-detection.
+    "SYSTEMD=1"
+  ];
 
   meta = with lib; {
     description = "A haskell-written notification center for users that like a desktop with style";
```

Though ideally, we would build it from scratch.

Reply:
Yeah that is much cleaner, thanks for the suggestion, I applied the patch.

I would still like to build from scratch, the main problem is broken dependencies in haskellPackages.